src/main/java/org/codehaus/mojo/servicedocgen/ServiceOpenApi.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.codehaus.mojo.servicedocgen;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.codehaus.mojo.servicedocgen.descriptor.ServicesDescriptor;
+
+/**
+ * {@link AbstractMojo Maven Plugin} to automatically generate documentation for services of the current project.
+ * <ol>
+ * <li>Scans the current projects source code for (JAX-RS annotated) services that match the RegEx configured by
+ * <code>classnameRegex</code>.</li>
+ * <li>Analyzes the services from source-code (extract JavaDoc, etc.) and byte-code (resolve generic parameters, etc.)
+ * and create intermediate meta-data as {@link ServicesDescriptor}.</li>
+ * <li>Generates documentation from the collected meta-data (by default as HTML from a velocity template shipped with
+ * this plugin but can be overridden via configuration parameters).</li>
+ * </ol>
+ * scan and analyze services from the current project via source-code and byte-code analysis. Creates
+ *
+ * @author hohwille
+ */
+@Mojo(name = "openapi", defaultPhase = LifecyclePhase.PREPARE_PACKAGE, requiresProject = true, requiresDirectInvocation = false, executionStrategy = "once-per-session", requiresDependencyCollection = ResolutionScope.COMPILE_PLUS_RUNTIME)
+public class ServiceOpenApi extends ReportGen {
```

Comment:
Maven naming conventions would be nice. Can you rename the class to `ServiceOpenApiReport` please.